sources/web/datalab/polymer/modules/bigquery-file-manager/bigquery-file-manager.ts

```diff
@@ -92,19 +99,31 @@ class BigQueryFileManager implements FileManager {
     throw new UnsupportedMethod('getEditorUrl', this);
   }
 
+  private async _listMoreProjects(accumulatedProjects: ProjectResource[],
```

Comment:
Do we want to name this function differently so it can work with projects/datasets/tables? Maybe pass in the type of objects or the actual API to call?

Reply:
I thought about that ... but based on YAGNI I decided to wait until we actually have a project with more than 1000 datasets or a dataset with more than 1000 tables.
If you think it is worth doing this now, I can take another look at it.

Reply:
All hail YAGNI, but I think this is something any user might hit, so we can't wait until this happens to us.
If you don't want to put it as part of this PR, maybe put a hi-pri TODO?

Reply:
I started trying to make this function generic, but there were enough differences that I decided to just write the equivalent methods for listDatasets and listTables. PTAL.